File: benchmark/src/php/exception_ladder.php
<?php
function f3($i){if($i%7==0)throw new Exception("boom");return$i*2;}
function f2($i){return f3($i);}
function f1($i){return f2($i);}
$N=200000;$acc=$caught=0;$start=hrtime(true);
for($i=1;$i<=$N;$i++){try{$acc=($acc+f1($i))%1000003;}catch(Exception$e){$caught++;}}
$end=hrtime(true);$ms=($end-$start)/1e6;echo"Result: {$acc}_{$caught}\n";echo"Time: ".round($ms)."ms\n";

File: benchmarks/src/ruby/exception_ladder.rb
def f3(i);raise "boom" if i%7==0;i*2;end
def f2(i);f3(i);end
def f1(i);f2(i);end
N=200000;acc=caught=0;start=Time.now.to_f*1000
(1..N).each{|i|begin;acc=(acc+f1(i))%1000003;rescue;caught+=1;end}
finish=Time.now.to_f*1000;puts "Result: #{acc}_#{caught}";puts "Time: #{(finish-start).round}ms"
